src/main/java/com/michelin/ns4kafka/services/executors/TopicAsyncExecutor.java

```diff
@@ -391,36 +473,62 @@ private void createAndAssociateTags(List<TagTopicInfo> tagsToAssociate) {
+                                    .anyMatch(name -> topic.getMetadata().getName().equals(name)))
+                                .forEach(topic -> {
+                                    topic.getMetadata().setGeneration(topic.getMetadata().getGeneration() + 1);
+                                    topic.setStatus(Topic.TopicStatus.ofSuccess("Topic tags updated"));
+                                });
+                            log.info(String.format("Success associating tag %s.", stringTags));
+                        },
+                        error -> {
+                            ns4kafkaTopics.stream()
+                                .filter(topic -> tagsToAssociate
+                                    .stream()
+                                    .map(TagTopicInfo::entityName)
+                                    .map(name -> name.substring(name.lastIndexOf(":") + 1))
+                                    .anyMatch(name -> topic.getMetadata().getName().equals(name)))
+                                .forEach(topic -> topic.setStatus(Topic.TopicStatus.ofFailed(
+                                    "Error while associating topic tags:" + error.getMessage())));
+                            log.error(String.format("Error associating tag %s.", stringTags), error);
+                        }),
                 error -> log.error(String.format("Error creating tag %s.", stringTags), error));
     }
 
     /**
      * Dissociate tags to a topic.
      *
      * @param tagsToDissociate The tags to dissociate
-     * @param topicName        The topic name
+     * @param topic            The topic
      */
-    private void dissociateTags(Collection<String> tagsToDissociate, String topicName) {
+    private void dissociateTags(Collection<String> tagsToDissociate, Topic topic) {
         tagsToDissociate
             .forEach(tag -> schemaRegistryClient.dissociateTag(managedClusterProperties.getName(),
                     managedClusterProperties.getConfig().getProperty(CLUSTER_ID)
-                        + ":" + topicName, tag)
-                .subscribe(success -> log.info(String.format("Success dissociating tag %s.",
-                        managedClusterProperties.getConfig().getProperty(CLUSTER_ID) + ":"
-                            + topicName
-                            + "/" + tag)),
-                    error -> log.error(String.format("Error dissociating tag %s.",
-                        managedClusterProperties.getConfig().getProperty(CLUSTER_ID) + ":"
-                            + topicName
-                            + "/" + tag), error)));
+                        + ":" + topic.getMetadata().getName(), tag)
+                .subscribe(
+                        success -> {
+                            topic.getMetadata().setGeneration(topic.getMetadata().getGeneration() + 1);
+                            topic.setStatus(Topic.TopicStatus.ofSuccess("Topic tags updated"));
```

Comment:
"Topic tags dissociated"

Reply:
Right

Reply:
OK for that

`topicRepository.create(topic);` is missing to update the topic in the internal topic

Reply:
Right